app/models/claim/matching_attribute_finder.rb

```diff
@@ -43,9 +43,9 @@ def matching_claims
 
         concatenated_columns = "CONCAT(#{attributes.join(",")})"
         policies_to_find_matches.map { |policy|
-          policy::Eligibility.where("LOWER(#{concatenated_columns}) = LOWER(?)", vals.join)
+          policy::Eligibility.where("LOWER(#{concatenated_columns}) = LOWER(?)", vals.join) if (attributes - policy::Eligibility.column_names).empty?
```

Comment:
🤔 could this result in a missed match? Perhaps it's better to strip out any columns which don't exist but still match those which do, rather than skipping entirely.

Reply:
So right now, there is never more than one column, so we probably don't need to concatenate here.
I can refactor this code to remove the concat - not sure why it's here.
The column is based on the `ELIGIBILITY_MATCHING_ATTRIBUTES` for the policy,
which is `teacher_reference_number` for TSLR, LUPP, FE and ECP.
and `passport_number` for IRP.
I guess in theory, you could specify multiple columns per policy, but we haven't.

Reply:
To answer your question - I don't think this will result in a missed match.
That where clause is to check for eligibilities that match **all** of the columns in concatenated_columns.
If one of those columns does not exist on an eligibility, then there will never be a match with that eligibility.

Reply:
I think [this commit](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/3348/commits/13e17f0553ee118307f00a5c08c5ab40e401ccc5) should let us still do the comparison if the eligibilities have some different attributes

Reply:
oh yeah - that is better 👍🏽